src/component/Slider/Slider.jsx

```diff
@@ -0,0 +1,275 @@
+          ...scrollingState,
+          isTransition: true,
+          isScrolling: false,
+          scrollX: 0,
+          clientX: 0,
+        })
+        return;
+      }
+
+      onClickNext();
+    } else if (scrollingState.scrollX < -0.3) {
+      onClickPrev();
+    } else {
+      currentSlideShow();
+    }
+    setScrollingState({
+      ...scrollingState,
+      isTransition: true,
+      isScrolling: false,
+      scrollX: 0,
+      clientX: 0,
+    })
+  }
+
+  const currentSlideShow = () => {
+    setScrollingState({
+      ...scrollingState,
+      isTransition: true,
+      isScrolling: false,
+      scrollX: 0,
+    })
+  }
+
+  const onMouseUp = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+    changeSlideHandle();
+  }
+
+  const onTouchEnd = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    changeSlideHandle();
+  }
+
+  const onMouseLeave = e => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    e.preventDefault();
+    changeSlideHandle();
+  }
+
+  const onTouchCancel = () => {
+    if (!scrollingState.isScrolling) {
+      return;
+    }
+
+    changeSlideHandle();
+  }
+
+  const renderPrevSlide = () => {
+    const output = [];
+    for (let i = 0; i < slidesPerView; i += 1) {
+      output.push(children[length - 1 - i]);
+    }
+
+    return output.reverse();
+  };
+
+  const renderNextSlide = () => {
+    const output = [];
+    for (let i = 0; i < slidesPerView; i += 1) {
+      output.push(children[i]);
+    }
+
+    return output;
+  };
+
+  return (
+    <div 
+      className="carousel__container"
+      >
+      <div
+        ref={carouselRef}
+        className={`carousel__slides ${scrollingState.isTransition ? 'transition' : 'transition-stop'}`}
+        style={{
+          transform: `translateX(-${(currentIndex + scrollingState.scrollX) * 100}%)`,
+          width: `calc(100%/${slidesPerView})`,
+        }}
+        onTouchStart={onTouchStart}
+        onTouchMove={onTouchMove}
+        onTouchEnd={onTouchEnd}
+        onTouchCancel={onTouchCancel}
+        onMouseDown={onMouseDown}
+        onMouseUp={onMouseUp}
+        onMouseMove={onMouseMove}
+        onMouseLeave={onMouseLeave}
+        onTransitionEnd={handleTransitionEnd}
+      >
+        {
+          (length >= slidesPerView && infiniteLoop)
+          && renderPrevSlide()
+        }
+        {children}
+        {
+          (length >= slidesPerView && infiniteLoop)
+          && renderNextSlide()
+        }
+      </div>
+      {
+        !showProgressBar
+        || (
+        <ProgressBar
+          length={infiniteLoop ? length : slidesPerView === 1 ? length : length - 1}
+          slidesPerView={infiniteLoop ? slidesPerView : 0}
+          currentIndex={currentIndicator}
+          onClickIndicator={onClickIndicator}
+        />
+        )
+      }
+      {(infiniteLoop || currentIndex > 0) && <Button onClick={onClickPrev} className="btn__prev" icon={<Prev />} />}
```

Comment:
should be multiline
`onClickPrev` wrong. `onPrevClick` better. `handlePrevClick` amazing

Reply:
I named this function `handlePrevClick`